src/ui/MainMenu.cpp

```diff
@@ -6054,26 +7541,118 @@ namespace MainMenu {
+		    switch (index) {
+		    case 0: cancel->setCallback([](Button&){createCharacterCard(0);}); break;
+		    case 1: cancel->setCallback([](Button&){createCharacterCard(1);}); break;
+		    case 2: cancel->setCallback([](Button&){createCharacterCard(2);}); break;
+		    case 3: cancel->setCallback([](Button&){createCharacterCard(3);}); break;
+		    }
+		    cancel->select();
+		} else {
+		    auto status = card->addField("status", 64);
+		    if (ready) {
+		        status->setText("Ready!");
+		    } else {
+		        status->setText("Not Ready");
+		    }
+		    status->setFont(smallfont_outline);
+		    status->setSize(SDL_Rect{(card->getSize().w - 200) / 2, card->getSize().h / 2, 200, 50});
+		    status->setVJustify(Button::justify_t::TOP);
+		    status->setHJustify(Button::justify_t::CENTER);
 		}
-		cancel->select();
+
+        // determine if all players are ready, and if so, create a countdown timer
+		bool allReady = true;
+		for (int c = 0; c < 4; ++c) {
+			auto card = lobby->findFrame((std::string("card") + std::to_string(c)).c_str());
+			if (!card) {
+			    // this can happen when a player enters a lobby and not all the stones exist yet.
+			    allReady = false;
+			    break;
+			}
+			auto backdrop = card->findImage("backdrop"); assert(backdrop);
+			if (backdrop->path == "images/ui/Main Menus/Play/PlayerCreation/UI_Invite_Window00.png") {
+				playersInLobby[c] = false;
+				if (multiplayer != SINGLE && !client_disconnected[c]) {
+			        allReady = false;
+			    }
+			} else if (backdrop->path == "images/ui/Main Menus/Play/PlayerCreation/UI_Ready_Window00.png") {
+				playersInLobby[c] = true;
+			} else {
+				playersInLobby[c] = true;
+			    allReady = false;
+			}
+		}
+        if (local) {
+            sendReadyOverNet(index, ready);
+        }
+		if (allReady) {
+		    createCountdownTimer();
+		} else {
+	        auto countdown = lobby->findFrame("countdown");
+	        if (countdown) {
+	            countdown->removeSelf();
+	        }
+		}
+	}
+
+	static void startGame() {
+        destroyMainMenu();
+        createDummyMainMenu();
+	    if (multiplayer == CLIENT) {
+            beginFade(MainMenu::FadeDestination::GameStartDummy);
+	    } else {
+            beginFade(MainMenu::FadeDestination::GameStart);
+
+	        // initialize all player stats
+	        for (int c = 1; c < MAXPLAYERS; c++) {
+		        if (!client_disconnected[c]) {
+			        if (!loadingsavegame || !intro) {
+				        stats[c]->clearStats();
+				        initClass(c);
+			        } else {
+				        loadGame(c);
+			        }
+		        }
+	        }
+
+	        // set unique game key
+	        uniqueGameKey = prng_get_uint();
+	        if (!uniqueGameKey) {
+		        uniqueGameKey++;
+	        }
+
+	        // send start signal to each player
+	        if (multiplayer == SERVER) {
+	            for (int c = 1; c < MAXPLAYERS; c++) {
+		            if (client_disconnected[c] || players[c]->isLocalPlayer()) {
```

Comment:
Can't use `players[c]->isLocalPlayer()` here as `intro` is true and every client returns true on the lobby screen

Reply:
fair enough, will just change it to `c == clientnum`